src/Functions/array/arrayResize.cpp

```diff
@@ -21,40 +22,36 @@ namespace ErrorCodes
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
 }
 
-DataTypePtr FunctionArrayResize::getReturnTypeImpl(const DataTypes & arguments) const
+DataTypePtr FunctionArrayResize::getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const
 {
-    const size_t argument_count = arguments.size();
 
-    if (argument_count < 2 || argument_count > 3)
```

Comment:
Ah, so `ColumnsWithTypeAndName` along with defining `mandatory_args` and `optional_args` is the more "modern" implementation of functions? 👍🏼

Reply:
Yeah, that's the 21st century way of parsing the arguments :-)